stan/math/prim/mat/fun/gp_dot_prod_cov.hpp

```diff
@@ -2,12 +2,16 @@
 #define STAN_MATH_PRIM_MAT_FUN_COV_DOT_PROD_HPP
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
-#include <stan/math/prim/mat/fun/dot_product.hpp>
-#include <stan/math/prim/mat/fun/value_of.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
 #include <stan/math/prim/scal/err/check_nonnegative.hpp>
 #include <stan/math/prim/scal/err/check_not_nan.hpp>
 #include <stan/math/prim/scal/meta/return_type.hpp>
+#include <stan/math/prim/mat/fun/dot_product.hpp>
+#include <stan/math/prim/mat/fun/dot_self.hpp>
+#include <boost/utility/enable_if.hpp>
```

Comment:
Once you change the function signatures, you can get rid of some of these new headers.

Reply:
Hey - I still need these two function signatures. There are two cases: a 1D GP, where the dot product is element wise multiplication (dot_self). D>1 => I need dot_product... is this clear enough?